content/docs/legacy/application-services.md

```diff
@@ -0,0 +1,225 @@
+- The bot can trivially manage hundreds of users. Events are pushed to the
+  application service directly. If you tried to do this as a client-side bot,
+  you would need one event stream connection per virtual user.
+- The bot can lazily create rooms on demand. This means Matrix users can join
+  non-existent room aliases and have the application service quickly track an
+  IRC channel and create a room with that alias, allowing the join request to
+  succeed.
+
+### Implementation details:
+
+- Written in Node.js, designed to be run using forever.
+- Built on the generic [matrix-appservice-node](https://github.com/matrix-org/matrix-appservice-node)
+  framework.
+- Supports sending metrics in statsd format.
+- Uses matrix-appservice-node to provide a standardised interface when writing
+  application services, rather than an explicit web framework (though under the
+  hood matrix-appservice-node is using Express).
+
+At present, the IRC application service is in beta, and is being run on #matrix
+and #matrix-dev. If you want to give it a go, [check it out on Github ]
+(https://github.com/matrix-org/matrix-appservice-irc)!
+
+## What Application services can do for you
+
+Application services have enormous potential for creating new and exciting ways
+to transform and enhance the core Matrix protocol. For example, you could
+aggregate information from multiple rooms into a summary room, or create
+throwaway virtual user accounts to proxy messages for a fixed user ID
+on-the-fly. As you may expect, all of this power assumes a high degree of trust
+between application services and homeservers. Only homeserver admins can allow
+an application service to link up with their homeserver, and the application
+service is in no way federated to other homeservers. You can think of
+application services as additional logic on the homeserver itself, without
+messing around with the book-keeping that homeservers have to do. This makes
+adding useful functionality very easy.
+
+### Example
+
+The application service (AS) API itself uses webhooks to communicate from the
+homeserver to the AS:
+
+- Room Alias Query API : The homeserver hits a URL on your application server to
+  see if a room alias exists.
+- User Query API : The homeserver hits a URL on your application server to see
+  if a user ID exists.
+- Push API : The homeserver hits a URL on your application server to notify you
+  of new events for your users and rooms.
+
+A very basic application service may want to log all messages in rooms which
+have an alias starting with "#logged_" (side note: logging won't work if these
+rooms are using end-to-end encryption).
+
+Here's an example of a very basic application service using Python (with Flask
+and Requests) which logs room activity:
+
+```python
+# app_service.py:
+
+import json, requests  # we will use this later
+from flask import Flask, jsonify, request
+app = Flask(__name__)
+
+@app.route("/transactions/&lt;transaction&gt;", methods=["PUT"])
```

Comment:
This `&lt;` / `&gt;` thing seems wrong, was this a non-markdown code block before?